atomic_reactor/utils/imageutil.py

```diff
@@ -92,33 +100,76 @@ def _get_registry_client(self, registry: str) -> util.RegistryClient:
         session = util.RegistrySession.create_from_config(self._conf, registry)
         return util.RegistryClient(session)
 
+    def extract_file_from_image(self, image: str, src_path: str, dst_path: str) -> None:
+        """Extract file or directory from image at src_path to dst_path
 
-# OSBS2 TBD
-
-
-def get_image_history(image):
-    # get image history with skopeo / registry api
-    return []
-
-
-def inspect_built_image():
-    # get output image final/arch specific from somewhere
-    # and call get_inspect_for_image
-    return {}
-
-
-def remove_image(image, force=False):
-    # self.tasker.remove_image(image, force=force)
-    # most likely won't be needed at all
-    return {}
-
+        :param image: str, image pullspec from which to extract
+        :param src_path: str, path inside the image that points to file or directory
+                             that will be extracted
+        :param dst_path: str, path where to export file/dir
+        """
+        cmd = ['oc', 'image', 'extract', f'{image}', '--path', f'{src_path}:{dst_path}']
 
-def tag_image(image, new_image):
-    # self.tasker.tag_image(image, new_image)
-    return True
+        try:
+            retries.run_cmd(cmd)
+        except subprocess.CalledProcessError as e:
+            logger.error("Image file extraction failed\n%s", e.output)
+            raise
 
+    def download_image_archive_tarball(self, image: str, path: str) -> None:
+        """Downloads image archive tarball to path.
 
-def get_image(image):
-    # self.tasker.get_image(image)
-    # use skopeo copy
-    return {}
+        :param image: str, image pullspec to download
+        :param path: str, path including the filename of the tarball
+        """
+        cmd = ['skopeo', 'copy', f'docker://{image}', f'docker-archive://{path}']
+        try:
+            retries.run_cmd(cmd)
+        except subprocess.CalledProcessError as e:
+            logger.error("Image archive download failed:\n%s", e.output)
+            raise
+
+    def get_uncompressed_image_layer_sizes(self, path: str) -> List[Dict[str, object]]:
+        """Returns data about the uncompressed image layer sizes
+
+        :param path: str path to a image archive tarball
+        :return: List[Dict[str, object]], where key is layer id and int is the size
+                                 in bytes
```

Comment:
If it's not possible to further specify the value type, it should be `Any` rather than `object`. Not sure if there are any practical differences, but `Any` is the idiomatic way to leave the type unspecified.

And the description of the return value appears outdated

Reply:
fixed